File: sample-java/src/main/java/com/smart/github/app/trend/TrendAdapter.java
package com.smart.github.app.trend;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;
import androidx.annotation.NonNull;
import butterknife.BindView;
import com.compact.widget.CompactRecyclerView;
import com.smart.github.domain.model.Trend;

public class TrendAdapter extends CompactRecyclerView.Adapter<Trend, TrendAdapter.ViewHolder> {

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        return new ViewHolder(
                LayoutInflater.from(parent.getContext()).inflate(android.R.layout.simple_list_item_1, parent, false)
        );
    }

    public class ViewHolder extends CompactRecyclerView.ViewHolder<Trend> {

        @BindView(android.R.id.text1)
        TextView name;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
        }

        @Override
        protected void bind(int position, Trend object) {
            name.setText(object.getFullName());
        }
    }
}

File: sample/src/main/java/com/smart/github/app/trend/TrendAdapter.kt
package com.smart.github.ui.trend

import android.R
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.compact.widget.CompactRecyclerView
import com.smart.github.domain.model.Trend

class TrendAdapter : CompactRecyclerView.Adapter<Trend, TrendAdapter.ViewHolder>() {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        return ViewHolder(
                LayoutInflater.from(parent.context).inflate(
                        R.layout.simple_list_item_1,
                        parent,
                        false
                )
        )
    }

    class ViewHolder(itemView: View) : CompactRecyclerView.ViewHolder<Trend>(itemView) {
        private val name: TextView = itemView.findViewById(android.R.id.text1)

        override fun bind(position: Int, obj: Trend?) {
            name.text = obj?.fullName
        }

    }

}
